tests/components/onedrive/test_backup.py

```diff
@@ -0,0 +1,326 @@
+"""Test the backups for OneDrive."""
+
+from __future__ import annotations
+
+from collections.abc import AsyncGenerator
+from html import escape
+from io import StringIO
+from json import dumps
+from unittest.mock import Mock, patch
+
+from kiota_abstractions.api_error import APIError
+from msgraph.generated.models.drive_item import DriveItem
+import pytest
+
+from homeassistant.components.backup import DOMAIN as BACKUP_DOMAIN, AgentBackup
+from homeassistant.components.onedrive.const import DOMAIN
+from homeassistant.config_entries import SOURCE_REAUTH
+from homeassistant.core import HomeAssistant
+from homeassistant.setup import async_setup_component
+
+from . import setup_integration
+from .const import BACKUP_METADATA
+
+from tests.common import AsyncMock, MockConfigEntry
+from tests.typing import ClientSessionGenerator, MagicMock, WebSocketGenerator
+
+
+@pytest.fixture(autouse=True)
+async def setup_backup_integration(
+    hass: HomeAssistant,
+    mock_config_entry: MockConfigEntry,
+) -> AsyncGenerator[None]:
+    """Set up onedrive integration."""
+    with (
+        patch("homeassistant.components.backup.is_hassio", return_value=False),
+        patch("homeassistant.components.backup.store.STORE_DELAY_SAVE", 0),
+    ):
+        assert await async_setup_component(hass, BACKUP_DOMAIN, {BACKUP_DOMAIN: {}})
+        await setup_integration(hass, mock_config_entry)
+
+        await hass.async_block_till_done()
+        yield
+
+
+async def test_agents_info(
+    hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test backup agent info."""
+    client = await hass_ws_client(hass)
+
+    await client.send_json_auto_id({"type": "backup/agents/info"})
+    response = await client.receive_json()
+
+    assert response["success"]
+    assert response["result"] == {
+        "agents": [
+            {"agent_id": "backup.local"},
+            {"agent_id": f"{DOMAIN}.{mock_config_entry.title}"},
+        ],
+    }
+
+
+async def test_agents_list_backups(
+    hass: HomeAssistant,
+    hass_ws_client: WebSocketGenerator,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test agent list backups."""
+
+    client = await hass_ws_client(hass)
+    await client.send_json_auto_id({"type": "backup/info"})
+    response = await client.receive_json()
+
+    assert response["success"]
+    assert response["result"]["agent_errors"] == {}
+    assert response["result"]["backups"] == [
+        {
+            **AgentBackup.from_dict(BACKUP_METADATA).as_frontend_json(),
```

Comment:
Can we hardcode the expected output dict? 
I don't think we should use internal methods to generate the expected output of a test.